tempfile `3.17` compile failed while using `windows-sys 0.52`

Hi, @Stebalien, thank you for the maintenance. I found that the latest `tempfile` 3.17 failed to compile when using `windows-sys` 0.52.

The error message:

```
error[E0308]: mismatched types
   --> C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tempfile-3.17.0\src\file\imp\windows.rs:34:13
    |
33  |         if SetFileInformationByHandle(
    |            -------------------------- arguments to this function are incorrect
34  |             f.as_raw_handle(),
    |             ^^^^^^^^^^^^^^^^^ expected `isize`, found `*mut c_void`
    |
    = note:     expected type `isize`
            found raw pointer `*mut c_void`
note: function defined here
   --> C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\windows-sys-0.52.0\src\Windows\Win32\Storage\FileSystem\mod.rs:607:106
    |
607 | ::windows_targets::link!("kernel32.dll" "system" #[doc = "Required features: `\"Win32_Foundation\"`"] fn SetFileInformationByHandle(hfile...
    |                                                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `tempfile` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

The commit that breaks the build: https://github.com/Stebalien/tempfile/commit/461369fd358b374040ef99e1cdb260eb5dcd1f85

--- a/src/file/imp/windows.rs
+++ b/src/file/imp/windows.rs
@@ -31,7 +31,7 @@ fn delete_open_file(f: &File) -> io::Result<()> {
             Flags: FILE_DISPOSITION_FLAG_DELETE | FILE_DISPOSITION_FLAG_POSIX_SEMANTICS,
         };
         if SetFileInformationByHandle(
-            f.as_raw_handle(),
+            f.as_raw_handle() as HANDLE,
             FileDispositionInfoEx,
             &info as *const _ as *const _,
             std::mem::size_of::<FILE_DISPOSITION_INFO_EX>() as u32,

